app/src/test/java/org/mozilla/vrbrowser/GleanMetricsServiceTest.kt

```diff
@@ -0,0 +1,143 @@
+package org.mozilla.vrbrowser
+
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.service.glean.testing.GleanTestRule
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.vrbrowser.GleanMetrics.*
+import org.mozilla.vrbrowser.telemetry.GleanMetricsService
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class GleanMetricsServiceTest {
+
+    @get:Rule
+    val gleanRule = GleanTestRule(ApplicationProvider.getApplicationContext())
+
+    @Test
+    fun testURLTelemetry() {
+        assertFalse(Url.domains.testHasValue())
+        assertFalse(Url.visits.testHasValue())
+        GleanMetricsService.stopPageLoadTimeWithURI("https://www.example.com/test.html");
+        assertTrue(Url.domains.testHasValue())
+        assertEquals(Url.domains.testGetValue(), 1)
+
+        assertTrue(Url.visits.testHasValue())
+        assertEquals(Url.visits.testGetValue(), 1)
+
+        assertFalse(Url.queryType["type_link"].testHasValue())
+        assertFalse(Url.queryType["type_query"].testHasValue())
+        GleanMetricsService.urlBarEvent(true)
+        assertTrue(Url.queryType["type_link"].testHasValue())
+        assertEquals(Url.queryType["type_link"].testGetValue(), 1)
+        GleanMetricsService.urlBarEvent(false)
+        assertTrue(Url.queryType["type_query"].testHasValue())
+        assertEquals(Url.queryType["type_query"].testGetValue(), 1)
+
+        assertFalse(Url.queryType["voice_query"].testHasValue())
+        GleanMetricsService.voiceInputEvent()
+        assertTrue(Url.queryType["voice_query"].testHasValue())
+        assertEquals(Url.queryType["voice_query"].testGetValue(), 1)
+    }
+
+    @Test
+    fun testDistributionChannelTelemetry() {
+        assertFalse(Distribution.channelName.testHasValue())
+        GleanMetricsService.testSetStartupMetrics()
+        assertTrue(Distribution.channelName.testHasValue())
+        assertEquals(Distribution.channelName.testGetValue(), BuildConfig.FLAVOR_platform)
+    }
+
+    @Test
+    fun testFxAAccountTelemetry() {
+        assertFalse(FirefoxAccount.signIn.testHasValue())
+        GleanMetricsService.FxA.signIn()
+        assertTrue(FirefoxAccount.signIn.testHasValue())
+        var events = FirefoxAccount.signIn.testGetValue()
+        assertEquals(events.size, 1)
+        assertEquals(events[0].category, "firefox_account")
+        assertEquals(events[0].name, "sign_in")
```

Comment:
Note: these checks will always be true because they will always come from `FirefoxAccount.signIn`. You can safely remove these.

See [the docs](https://mozilla.github.io/glean/book/user/metrics/event.html) for more details.